File: 프로그래머스/lv2/12909. 올바른 괄호/올바른 괄호.py
from collections import deque
def solution(s):
    answer = True
    stk = deque()
    for c in s:
        if c == '(':
            stk.append(c)
        else:
            if len(stk) > 0 and c == ')':
                stk.pop()
            else:
                answer = False
                break
    if len(stk) == 0 and answer:
        return True
    return False

File: 프로그래머스/lv2/12909. 올바른 괄호/올바른 괄호.java
import java.util.*;

class Solution {
    boolean solution(String s) {
        Stack<Character> stack = new Stack<>();
        for(int i = 0; i < s.length(); ++i){
            char now = s.charAt(i);
            if(now == '(')
                stack.push(now);
            else{
                if(stack.isEmpty())
                    return false;
                char top = stack.pop();
                if(top != '(')
                    return false;
            }
                
        }
        if(!stack.isEmpty())
            return false;
        return true;
    }
}